src/charm.py

```diff
@@ -108,6 +112,30 @@ def _configure(self):
 
         self.unit.status = ActiveStatus()
 
+    def _set_alerts(self, container):
+        """Create alert rule files for all Prometheus consumers.
+
+        Args:
+            container: the Prometheus workload container into which
+                alert rule files need to be created.
+        """
+        if not self.provider_ready:
+            return
+
+        container.remove_path(RULES_DIR, recursive=True)
+
+        for rel_id, alerts in self.prometheus_provider.alerts().items():
+            filename = "juju_{}_{}_{}_rel_{}_alert.rules".format(
+                alerts["model"], alerts["model_uuid"], alerts["application"], rel_id
+            )
+            try:
```

Comment:
Replace this try/except and use `with container.is_ready():`?

Reply:
Nice. Thanks. Will do. Container `is_ready` seems new and was too busy to keep myself updated. It will be good if there was a way to get automated notifications when [sdk docs](https://juju.is/docs/sdk) or [ops docs](https://ops.rtfd.io) change.